source/background.ts

```diff
@@ -51,9 +52,30 @@ browser.browserAction.onClicked.addListener(async () => {
 	});
 });
 
+async function hasUsedStorage(): Promise<boolean> {
+	return (
+		await browser.storage.sync.getBytesInUse() > 0
+		// Note: Not available in Firefox https://bugzilla.mozilla.org/show_bug.cgi?id=1385832
+		|| Number(await browser.storage.local.getBytesInUse?.()) > 0
+	);
+}
+
+async function isFirstInstall(suggestedReason: string): Promise<boolean> {
+	return (
+		// Always exclude local installs from the welcome screen
+		!isDevelopmentVersion()
```

Comment:
I tested this in Chrome with and without this line and it seems that the logic is still sound at least there